docker-compose.yml

```diff
@@ -1,65 +1,109 @@
 version: '3.9'
 
-x-base_service: &base_service
-    ports:
-      - "7860:7860"
-    volumes:
-      - &v1 ./data:/data
-      - &v2 ./output:/output
-    stop_signal: SIGINT
-    deploy:
-      resources:
-        reservations:
-          devices:
-              - driver: nvidia
-                device_ids: ['0']
-                capabilities: [gpu]
+x-base_service:
+  &base_service
+  ports:
+    - "7860:7860"
+  volumes:
+    - &v1 ./data:/data
+    - &v2 ./output:/output
+  stop_signal: SIGINT
+  deploy:
+    resources:
+      reservations:
+        devices:
+          - driver: nvidia
+            device_ids: [ '0' ]
+            capabilities: [ gpu ]
+
+x-base_service_amd:
+  &base_service_amd
+  ports:
+    - "7860:7860"
+  volumes:
+    - &v1 ./data:/data
+    - &v2 ./output:/output
+  stop_signal: SIGINT
+  group_add:
+    - video
+  devices:
+    - "/dev/dri"
+    - "/dev/kfd"
```

Comment:
I was looking at getting this running within a docker container on windows but it seems like this does not exist on WSL yet and fails when I get to the last stage of running it.
```
Error response from daemon: error gathering device information while adding custom device "/dev/kfd": no such file or directory
```
"/dev/kfd: the main compute interface shared by all GPUs"  from https://rocmdocs.amd.com/_/downloads/en/latest/pdf/

I might try getting this working on linux and open another merge request.

Reply:
so it's unavailable for wsl + docker to use AMD gpus yet, right? @mpitropov any updates?

Reply:
I was unable to get it working for wsl + docker on AMD. I made my own version of this branch for linux and AMD for AUTOMATIC111 by looking at their repository. I don't know a clean way of creating the code such that I should make a merge request since there are no if statements in docker files to run AMD or NVidia specific code.

Reply:
@mtthw-meyer any thoughts? can your docker access the `/dev/kfd` device?